lib/react_on_rails/helper.rb

```diff
@@ -361,6 +391,32 @@ def build_react_component_result_for_server_rendered_string(
       prepend_render_rails_context(result)
     end
 
+    def build_react_component_result_for_server_streamed_content(
+      rendered_html_stream: required("rendered_html_stream"),
+      component_specification_tag: required("component_specification_tag"),
+      render_options: required("render_options")
+    )
+      content_tag_options_html_tag = render_options.html_options[:tag] || "div"
+      # The component_specification_tag is appended to the first chunk
+      # We need to pass it early with the first chunk because it's needed in hydration
+      # We need to make sure that client can hydrate the app early even before all components are streamed
+      is_first_chunk = true
+      rendered_html_stream = rendered_html_stream.transform do |chunk|
```

Comment:
The [StreamRequest](https://github.com/shakacode/react_on_rails_pro/pull/407/files#diff-c1d07594e0987e090d78fe43f67ee57bd751c72c4e5cec3df2ff04a8d2606c14R58) and [StreamDecorator](https://github.com/shakacode/react_on_rails_pro/pull/407/files#diff-c1d07594e0987e090d78fe43f67ee57bd751c72c4e5cec3df2ff04a8d2606c14R2) classes allow us to transform the stream of http body chunks